### PR TITLE
Tensorflow: update dense-train-mnist and predict-mnist

### PR DESCRIPTION
**Pull request by DrewParallel**
Tested manually of pm_docker_runner 1.0.10

### CODE REVIEW

File: components/Python/TensorFlow/predict-mnist/mnist_inference.py [hunk 1]
```diff
@@ -64,7 +64,8 @@ def main():
     if os.path.isdir(args.model_dir):
         print("Found model_dir")
     else:
-        raise ValueError("model dir is not found")
```
Comment (yakov-g) [inline components/Python/TensorFlow/predict-mnist/mnist_inference.py:67]: Are you sure we want to exit? Usually it is not a good practice to exit, exception would be better.

Comment (DrewParallel) [reply components/Python/TensorFlow/predict-mnist/mnist_inference.py:67]: Yes, in cases where the inference pipeline is waiting for a model from the training pipeline.
This is the standard most of the inference pipelines use.

**Review by yakov-g — APPROVED** [review]

File: components/Python/TensorFlow/predict-mnist/mnist_inference.py (hunk 1 above)
Comment (liororama) [reply components/Python/TensorFlow/predict-mnist/mnist_inference.py:67]: Actually this is a good time to fix this.. an inference pipeline should fail if there is no model provided.
Either a default model should be provided, or use the new (soon) mlapp triggering option.

**Review by liororama — APPROVED** [review]

File: components/Python/TensorFlow/predict-mnist/mnist_inference.py (hunk 1 above)
Comment (yakov-g) [reply components/Python/TensorFlow/predict-mnist/mnist_inference.py:67]: In argparse you  can define a callback

```python
class CompRootDirCheck(argparse.Action):
    def __call__(self, parser, namespace, values, option_string=None):
        comp_root_dir = values
        if not os.path.isdir(comp_root_dir):
            raise argparse.ArgumentTypeError("--comp-root argument value '{}' has to be existing dir".format(comp_root_dir))
        if os.access(comp_root_dir, os.R_OK):
            setattr(namespace, self.dest, comp_root_dir)
        else:
            raise argparse.ArgumentTypeError("--comp-root argument value '{}' is not a readable dir".format(comp_root_dir))
```

And assign it to argument

```python
deps.add_argument('-r', '--comp-root', default=None, required=True, action=CompRootDirCheck,
                                help='MLPiper components root dir. Recursively detecting components')
```

**Pull request merged**